zstash/extract.py

```diff
@@ -139,7 +139,11 @@ def extract(keep_files=True):
     config.keep = bool(int(config.keep))
 
     # The command line arg should always have precedence
-    config.keep = args.keep
+    if args.hpss == 'none':
+        # If no HPSS is available, always keep the files.
+        config.keep = True
```

Comment:
This being set to `False` is what causes the tar file deletion. I suppose we could run `os.chmod` on the tar file if `config.keep == True` to make the tar file read-only in this case, but this if-block appears to be sufficient.

Reply:
Maybe add 'lower' to the if test just in case someone specifies hpss=None.

Reply:
Actually, since you are testing against 'none' other places in the code, we should make sure to lower capitalization of the command line argument.

Reply:
@golaz Can that be done in one file (i.e., `main.py`) or would I have to apply `.lower()` in all the python files representing commands?

I was thinking it would be good to simply have `args.hpss` default to the `None` type if the user doesn't specify `--hpss`. However, `chgrp` appears to be an obstacle to that idea because `hpss` is passed in by position rather than command line option; changing how `chgrp` is called would break backward compatibility.

Reply:
@forsyth2 : yes, it probably would be cleaner to do the lower() in main.py rather than everywhere it appears in an if statement. config.hpss could be set to `None` rather than a string "none".

Setting args.hpss to default to `None` is likely break other things. In particular, you can invoke zstash without `--hpss` and in that case zstash should use the path stored in the database. The idea was to avoid having to specify `--hpss` at every invocation:

```
zstash create --hpss=... ...
zstash update   # no need to repeat --hpss
```

The 'chgrp' option is rarely used (and apparently not mentioned in the documentation pages), so you can break backwards compatibility for it.